hooks/repo.go

```diff
@@ -142,3 +158,12 @@ func addHook[T any](hooks map[string]T, hook T, id string) (map[string]T, error)
 
 	return hooks, nil
 }
+
+func addModuleStageName(id string, stage string) {
+	if moduleStageNameCollector == nil {
+		moduleStageNameCollector = make(map[string][]string)
+	}
+
+	str := strings.SplitN(id, ".", 2)
+	moduleStageNameCollector[str[1]] = append(moduleStageNameCollector[str[1]], stage)
```

Comment:
Unfortunately, there is no guarantee that the module name will be unique across different modules without the vendor name. Thus, we can end up having similar metric names for different modules.
Therefore, I suggest not removing the vendor name.

Reply:
ok, I agree with that, let's make `moduleCode` in metric as `vendor-module`